### PR TITLE
Convert history commands to `Action2`

### LINKED ISSUE #150782: Command center: back/forward should be conditionally enabled
Testing #150396

The buttons for history navigation should probably leverage `canNavigateBack` and `canNavigateForward` context keys for proper enablement.

**PS:** this is probably more of a UX discussion to have, but I wonder if the go back / go forward buttons are too prominent now in the title bar. I see the case for quick open, but other buttons should maybe be configurable by the user or not be added by default. I believe we are close to then having a real toolbar with customizable entries in the title, which I think a lot of users ask for (https://github.com/microsoft/vscode/issues/41309).

### PR DESCRIPTION
**Pull request by bpasero**
This PR fixes #150782

### CODE REVIEW

**Comment by bpasero** [comment]
@jrieken will need your help here, it does not look like the `precondition` gets applied to the menus. I notice how I can register multiple menus from `Action2` directly, but I think I cannot define different icons per menu, so I think something is missing for me to address this?

File: src/vs/workbench/browser/parts/editor/editorActions.ts [hunk 1]
```diff
@@ -1258,39 +1263,57 @@ export class OpenLastEditorInGroup extends AbstractNavigateEditorAction {
...
-	constructor(
-		id: string,
-		label: string,
-		@IHistoryService private readonly historyService: IHistoryService
-	) {
-		super(id, label);
+	constructor() {
+		super({
+			id: NavigateForwardAction.ID,
+			title: { value: localize('navigateForward', "Go Forward"), original: 'Go Forward' },
```
Comment (jrieken) [inline src/vs/workbench/browser/parts/editor/editorActions.ts:1274]: Also use `mnemonicTitle` here...

File: src/vs/workbench/browser/parts/editor/editorActions.ts [hunk 2]
```diff
@@ -1258,39 +1263,57 @@ export class OpenLastEditorInGroup extends AbstractNavigateEditorAction {
...
+			f1: true,
+			precondition: ContextKeyExpr.has('canNavigateForward'),
+			keybinding: {
+				weight: KeybindingWeight.WorkbenchContrib,
+				win: { primary: KeyMod.Alt | KeyCode.RightArrow },
+				mac: { primary: KeyMod.WinCtrl | KeyMod.Shift | KeyCode.Minus },
+				linux: { primary: KeyMod.CtrlCmd | KeyMod.Shift | KeyCode.Minus }
+			}
```
Comment (jrieken) [inline src/vs/workbench/browser/parts/editor/editorActions.ts:1282]: Use `menu: [{...},{...}]'

**Comment by jrieken** [comment]
> @jrieken will need your help here, it does not look like the precondition gets applied to the menus. I notice how I can register multiple menus from Action2 directly,

You should use `menu` from `Action2` and it should work (unless the rendering menu ignores it). Generally all UI defining properties should be defined centrally (inside `Action2`) but I understand the touch bar special case. For that you would still need a lone MenuRegistery.append...-call.

**Comment by bpasero** [comment]
@jrieken updated. I had to leave the touch bar menu thing in for the different icon to use. If I had a way to set the icon for the menu registration, I could also inline that.

File: src/vs/workbench/browser/parts/editor/editorActions.ts [hunk 3]
```diff
@@ -1258,39 +1263,67 @@ export class OpenLastEditorInGroup extends AbstractNavigateEditorAction {
...
+			]
+		});
 	}
 
-	override async run(): Promise<void> {
-		await this.historyService.goForward(GoFilter.NONE);
+	run(accessor: ServicesAccessor): void {
+		const historyService = accessor.get(IHistoryService);
+
+		historyService.goForward(GoFilter.NONE);
```
Comment (jrieken) [inline src/vs/workbench/browser/parts/editor/editorActions.ts:1294]: Unsure if it matters but the `goForward/Back` calls aren't awaited anymore

Comment (bpasero) [reply src/vs/workbench/browser/parts/editor/editorActions.ts]: @jrieken true, fixed 👍

**Comment by bpasero** [comment]
🍻

**Review by jrieken — APPROVED** [review]

**Pull request merged**